### PR TITLE
pytest_docker_postgres: Add support for load_database

### PR DESCRIPTION
**Pull request by damngamerz**

### CODE REVIEW

**Comment by damngamerz** [comment]
@michaelbeaumont up for review. But don't merge this yet tests need to be amended.

**Comment by damngamerz** [comment]
@michaelbeaumont nevermind can be merged since we still have our old `db_engine`. It will not break things up in dataservice.

**Comment by michaelbeaumont** [comment]
it needs to be updated in every `Pipfile` too though

File: pytest_docker_postgres/__init__.py [hunk 1]
```diff
@@ -65,3 +79,18 @@ def db_engine(in_docker_compose, docker_services):
...
     drop_database(url)
+
+
+@pytest.fixture(scope="function")
+def schema_db_engine(db_engine, load_database):
+    connection = db_engine.connect()
+    path = f"{Path.home()}/{load_database}/"
+    sql_files_path = glob.glob(path + "**/*.sql", recursive=True)
+    #files should be sorted according to numbers
+    sql_files_path.sort()
```
Comment (michaelbeaumont) [inline pytest_docker_postgres/__init__.py]: better immutable
```
sql_files = sorted(glob.glob(path + "**/*.sql", recursive=True))
```

Comment (michaelbeaumont) [reply pytest_docker_postgres/__init__.py]: also use `Path` to do the concatenation

File: pytest_docker_postgres/__init__.py [hunk 2]
```diff
@@ -65,3 +79,18 @@ def db_engine(in_docker_compose, docker_services):
     create_database(url)
     yield create_engine(url)
     drop_database(url)
+
+
+@pytest.fixture(scope="function")
+def schema_db_engine(db_engine, load_database):
+    connection = db_engine.connect()
+    path = f"{Path.home()}/{load_database}/"
```
Comment (michaelbeaumont) [inline pytest_docker_postgres/__init__.py]: why use `Path.home`?

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: pytest_docker_postgres/__init__.py [hunk 3]
```diff
@@ -14,13 +16,25 @@ def pytest_addoption(parser):
         default="",
         help="Assume inside a docker network",
     )
+    parser.addoption(
+        "--load-database",
+        action="store",
+        default="",
+        help="Assume inside a docker network load a database from a sql folder",
```
Comment (michaelbeaumont) [inline pytest_docker_postgres/__init__.py]: update help string

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: pytest_docker_postgres/__init__.py (hunk 2 above)
Comment (damngamerz) [reply pytest_docker_postgres/__init__.py]: for testing :stuck_out_tongue:  but we don't need this actually

File: pytest_docker_postgres/__init__.py [hunk 4]
```diff
@@ -65,3 +78,15 @@ def db_engine(in_docker_compose, docker_services):
...
     yield create_engine(url)
     drop_database(url)
+
+
+@pytest.fixture(scope="function")
+def schema_db_engine(db_engine, load_database):
+    connection = db_engine.connect()
+    sql_files = sorted(glob.glob(os.path.join(load_database, "**/*.sql"), recursive=True))
+    for file_path in sql_files:
+        file = open(file_path)
```
Comment (michaelbeaumont) [inline pytest_docker_postgres/__init__.py]: ah didn't notice, use `with` here i think

File: pytest_docker_postgres/__init__.py [hunk 5]
```diff
@@ -65,3 +78,15 @@ def db_engine(in_docker_compose, docker_services):
...
+        trans = connection.begin()
```
Comment (michaelbeaumont) [inline pytest_docker_postgres/__init__.py]: i'm not sure why this is necessary though, in theory sqlalchemy is always in autocommit mode
also it should be `with` as well

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: pytest_docker_postgres/__init__.py [hunk 6]
```diff
@@ -82,11 +82,11 @@ def db_engine(in_docker_compose, docker_services):
...
-        trans = connection.begin()
-        connection.execute(text(file.read()))
-        trans.commit()
+    with db_engine.connect() as conn:
+        sql_files = sorted(glob.glob(os.path.join(load_database, "**/*.sql"), recursive=True))
+        for file_path in sql_files:
+            with open(file_path) as file:
+                with conn.begin() as transaction:
+                    conn.execute(text(file.read()))
+                    transaction.commit()
```
Comment (michaelbeaumont) [inline pytest_docker_postgres/__init__.py]: `commit` is handled by the `transaction` context manager

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: pytest_docker_postgres/__init__.py [hunk 7]
```diff
@@ -65,3 +78,14 @@ def db_engine(in_docker_compose, docker_services):
...
+
+
+@pytest.fixture(scope="function")
+def schema_db_engine(db_engine, load_database):
+    with db_engine.connect() as conn:
+        sql_files = sorted(glob.glob(os.path.join(load_database, "**/*.sql"), recursive=True))
+        for file_path in sql_files:
+            with open(file_path) as file:
+                with conn.begin():
+                    conn.execute(text(file.read()))
```
Comment (michaelbeaumont) [inline pytest_docker_postgres/__init__.py]:
```suggestion
                    conn.connection.cursor().execute(file.read())
```

and I think then you don't need the `transaction` anymore

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: pytest_docker_postgres/__init__.py (hunk 7 above)
Comment (damngamerz) [reply pytest_docker_postgres/__init__.py]: transactions are inevitable. :stuck_out_tongue_closed_eyes:

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**